spec/unit/remote_system_spec.rb

```diff
@@ -80,12 +80,23 @@
 
       it "uses sudo when necessary" do
         expect(Cheetah).to receive(:run).with(
-          "ssh", "machinery@remotehost", "sudo", "LC_ALL=C", "ls", "/tmp", privileged: true
+          "ssh", "machinery@remotehost", "sudo", "-n", "LC_ALL=C", "ls", "/tmp", privileged: true
         )
 
         remote_system.remote_user = "machinery"
         remote_system.run_command("ls", "/tmp", privileged: true)
       end
+
+      it "raises an exception if the user is not allowed to run sudo" do
+        expect(Cheetah).to receive(:run).with(
+          "ssh", "machinery@remotehost", "sudo", "-n", "LC_ALL=C", "ls", "/tmp", privileged: true
+        ).and_raise(Cheetah::ExecutionFailed.new(nil, 1, "", "sudo: a password is required"))
+
+        remote_system.remote_user = "machinery"
+        expect {
+          remote_system.run_command("ls", "/tmp", privileged: true)
+        }.to raise_error(Machinery::Errors::InsufficientPrivileges)
```

Comment:
I would check the error output too with since you don't check it anywhere else I think.